src/world_interface/real_world_interface.cpp

```diff
@@ -113,34 +113,26 @@ void initMotors() {
 	addMotorMapping(motorid_t::hand, false);
 }
 
-void setupCameras() {
+void openCamera(CameraID camID, const char* cameraPath) {
 	try {
 		auto cam = std::make_shared<cam::Camera>();
-		cam->openFromConfigFile(Constants::AR_CAMERA_CONFIG_PATH);
-		cameraMap[Constants::AR_CAMERA_ID] = cam;
-	} catch (const std::exception& e) {
-		log(LOG_ERROR, "Error opening camera with id %s:\n%s\n",
-			util::to_string(Constants::AR_CAMERA_ID).c_str(), e.what());
-	}
-	/* TODO: see why only 2 cams can be open at a time
-	try {
-		auto cam = std::make_shared<cam::Camera>();
-		cam->openFromConfigFile(Constants::FOREARM_CAMERA_CONFIG_PATH);
-		cameraMap[Constants::FOREARM_CAMERA_ID] = cam;
-	} catch (const std::exception& e) {
-		log(LOG_ERROR, "Error opening camera with id %s:\n%s\n",
-			util::to_string(Constants::AR_CAMERA_ID).c_str(), e.what());
-	}
-	*/
-	try {
-		auto cam = std::make_shared<cam::Camera>();
-		cam->openFromConfigFile(Constants::HAND_CAMERA_CONFIG_PATH);
-		cameraMap[Constants::HAND_CAMERA_ID] = cam;
+		bool success = cam->openFromConfigFile(cameraPath);
+		if (success) {
+			cameraMap[camID] = cam;
+		} else {
+			log(LOG_ERROR, "Failed to open camera with id %s\n",
+				util::to_string(camID).c_str());
+		}
 	} catch (const std::exception& e) {
 		log(LOG_ERROR, "Error opening camera with id %s:\n%s\n",
-			util::to_string(Constants::AR_CAMERA_ID).c_str(), e.what());
+			util::to_string(camID).c_str(), e.what());
 	}
 }
+
+void setupCameras() {
+	// openCamera(Constants::AR_CAMERA_ID, Constants::AR_CAMERA_CONFIG_PATH);
```

Comment:
Non-blocking: Just want to make sure this isn't meant to be called. The original code uses AR_CAMERA_ID but this one only uses HAND_CAMERA_ID.